document_merge_service/api/serializers.py

```diff
@@ -66,6 +66,14 @@ def _(doc):
         return sorted([x.replace(".[]", "[]") for x in _doc(sample_doc)])
 
     def validate(self, data):
+        user = self.context["request"].user
+
+        if self.instance is None:
+            data["created_by_user"] = user.username
+            data["created_by_group"] = user.group
+        data["modified_by_user"] = user.username
+        data["modified_by_group"] = user.group
```

Comment:
@open-dynaMIX Thanks for the pair implementation with @JohnL17! Quick question: I didn't quite understand which group is written into the created_by_group / modified_by_group fields, or better said where the data is coming from and how it relates to the user's group in eBau. Could you provide some insights? :)

Reply:
This comes from the IDP. We get a list of groups (strings), [accessible via `user.groups`](https://github.com/adfinis/document-merge-service/blob/main/document_merge_service/api/authentication.py#L34). We treat the first group as main group and [make it accessible via `user.group`](https://github.com/adfinis/document-merge-service/blob/main/document_merge_service/api/authentication.py#L18).

Reply:
Thanks for clarifying :+1: